src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -17,123 +21,170 @@ class MSMonitor(MSCore):
     MSMonitor class provide whole logic behind
     the transferor monitoring module.
     """
+    def __init__(self, msConfig, logger=None):
+        super(MSMonitor, self).__init__(msConfig, logger)
+        # update interval is used to check records in CouchDB and update them
+        # after this interval, default 6h
+        self.updateInterval = self.msConfig.get('updateInterval', 6*60*60)
+        self.campaigns = {}
+
+    def updateCaches(self):
+        """
+        Fetch some data required for the monitoring logic, e.g.:
+         * all campaign configuration
+        :return: True if all of them succeeded, else False
+        """
+        campaigns = self.reqmgrAux.getCampaignConfig("ALL_DOCS")
+        if not campaigns:
+            self.logger.warning("Failed to fetch the campaign configurations")
+        else:
+            # let's make campaign look-up easier and more efficient
+            self.campaigns = {}
+            for camp in campaigns:
+                if 'campaignName' not in camp:
+                    self.logger.warning(
+                            'No campaignName attribute in campaign dict: %s',
+                            json.dumps(camp))
+                    continue
+                self.campaigns[camp['campaignName']] = camp
+            return True
+        return False
+
     def execute(self, reqStatus):
         """
-        Executes the MS monitoring logic
+        Executes the MS monitoring logic, see
+        https://github.com/dmwm/WMCore/wiki/ReqMgr2-MicroService-Monitor
+
         :param reqStatus: request statue to process
         :return:
         """
+        if not self.updateCaches():
```

Comment:
and we better call `updateCaches` after getting the list of requests in `reqStatus`. So somewhere after this line:
https://github.com/dmwm/WMCore/pull/9325/files#diff-4cd93292a8b855fbad20e8e899adb8b7R72